File: app/src/main/java/io/github/zwieback/familyfinance/business/template/lifecycle/creator/TemplateCreator.java
package io.github.zwieback.familyfinance.business.template.lifecycle.creator;

import android.content.Context;

import java.util.TreeSet;

import io.github.zwieback.familyfinance.core.lifecycle.creator.EntityCreator;
import io.github.zwieback.familyfinance.core.model.Template;
import io.requery.Persistable;
import io.requery.reactivex.ReactiveEntityStore;

public class TemplateCreator extends EntityCreator<Template> {

    public TemplateCreator(Context context, ReactiveEntityStore<Persistable> data) {
        super(context, data);
    }

    @Override
    protected Iterable<Template> buildEntities() {
        return new TreeSet<>(this);
    }

    @Override
    public int compare(Template left, Template right) {
        return Integer.compare(left.getId(), right.getId());
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/template/lifecycle/creator/TemplateCreator.kt
package io.github.zwieback.familyfinance.business.template.lifecycle.creator

import android.content.Context
import io.github.zwieback.familyfinance.core.lifecycle.creator.EntityCreator
import io.github.zwieback.familyfinance.core.model.Template
import io.requery.Persistable
import io.requery.reactivex.ReactiveEntityStore

class TemplateCreator(
    context: Context,
    data: ReactiveEntityStore<Persistable>
) : EntityCreator<Template>(context, data) {

    override fun buildEntities(): Iterable<Template> {
        return sortedSetOf(this)
    }

    override fun compare(left: Template, right: Template): Int {
        return left.id.compareTo(right.id)
    }
}
